cartography/cli.py

```diff
@@ -734,6 +746,12 @@ def main(self, argv: str) -> int:
             config.semgrep_app_token = os.environ.get(config.semgrep_app_token_env_var)
         else:
             config.semgrep_app_token = None
+        # if config.semgrep_dependency_ecosystems:
+        #     # ideally we would store the result of this function to avoid having to call it again, but it doesn't seem
+        #     # like any existing config settings have a List type
+        #     parse_and_validate_semgrep_ecosystems(config.semgrep_dependency_ecosystems)
```

Comment:
I originally wanted to do the input validation here to match the existing `parse_and_validate_aws_requested_syncs` function, but realized that it would be breaking the pattern to have `config.semgrep_dependency_ecosystems` as a List instead of a str. And then I'd have to do the validation a second time in the dependencies module

Reply:
I'll pass this one to @achantavy.
Most of the configuration is done via environment variables as single string values.
There is [this case for Github credentials](https://cartography-cncf.github.io/cartography/modules/github/config.html?highlight=config) that uses a base64 encoded JSON, but I don't know if that would be the preferred approach here.

Reply:
Yeah I would copy the same thing with the --aws_requested_syncs param. Store it as a str, validate it without storing in cli.py, then validate + store it in the semgrep module. Slightly wasteful but it's only a small list.